File: templates/python/v1/factory/shape_factory.py
from abc import ABC, abstractmethod


class Shape(ABC):
    @abstractmethod
    def draw(self) -> str:
        raise NotImplementedError


class Circle(Shape):
    def draw(self) -> str:
        return "Drawing a Circle"


class Square(Shape):
    def draw(self) -> str:
        return "Drawing a Square"


class ShapeFactory:
    @staticmethod
    def create(kind: str) -> Shape:
        match kind.lower():
            case "circle":
                return Circle()
            case "square":
                return Square()
            case _:
                raise ValueError(f"Unknown shape: {kind}")

File: templates/javascript/v1/factory/shapeFactory.js
class Shape {
  draw() {
    throw new Error("Not implemented");
  }
}

class Circle extends Shape {
  draw() {
    return "Drawing a Circle";
  }
}

class Square extends Shape {
  draw() {
    return "Drawing a Square";
  }
}

export class ShapeFactory {
  static create(kind) {
    switch (kind.toLowerCase()) {
      case "circle":
        return new Circle();
      case "square":
        return new Square();
      default:
        throw new Error(`Unknown shape: ${kind}`);
    }
  }
}